File: Temp/leetcode/editor/cn/[1512]好数对的数目.py
# 1512 好数对的数目
# https://leetcode.cn/problems/number-of-good-pairs/
from imports import *


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        cnt = defaultdict(int)
        ans = 0
        for x in nums:
            ans += cnt[x]
            cnt[x] += 1
        return ans


class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        return sum(nums[i] == nums[j] for i in range(len(nums)) for j in range(i))
# leetcode submit region end(Prohibit modification and deletion)

File: Temp/leetcode/editor/cn/[1512]好数对的数目.c
// 1512 好数对的数目
// https://leetcode.cn/problems/number-of-good-pairs/


//leetcode submit region begin(Prohibit modification and deletion)
int numIdenticalPairs(int* nums, int numsSize) {
    int* cnt = (int*)malloc(sizeof(int) * 101);
    memset(cnt, 0, sizeof(int) * 101);
    int ans = 0;
    for (int i = 0; i < numsSize; i++) {
        ans += cnt[nums[i]]++;
    }
    free(cnt);
    return ans;
}
//leetcode submit region end(Prohibit modification and deletion)
